lib/nrf_modem_lib/nrf_modem_lib_trace.c

```diff
@@ -36,6 +36,54 @@ K_SEM_DEFINE(trace_done_sem, 1, 1);
 static int trace_init(void);
 static int trace_deinit(void);
 
+static double min_bps = DOUBLE_MAX;
+static double max_bps;
+static double tot_bps;
+static int num_bps;
```

Comment:
> I was wondering if there should be a kconfig to enable the extra calculations as well, not just the logging. The bps calculations adds extra complexity (and execution time) to the trace library. Although this is in the area of 10s of us per trace fragment, while it takes ~90us - 80ms to process each fragment, it will affect the total trace throughput slightly.

I thought about it and it would make the code much more verbose to little to no measurable improvement in performance. For example, the API function would only be available after toggling the option, which could create confusion. I'll see if I can find a better way to organize the code and please let me know if having a conditional API is a satisfying option and why.